frontend/widgets/OBSBasic_Transitions.cpp

```diff
@@ -198,17 +207,10 @@ obs_data_array_t *OBSBasic::SaveQuickTransitions()
 
 obs_source_t *OBSBasic::FindTransition(const char *name)
 {
-	for (int i = 0; i < ui->transitions->count(); i++) {
-		OBSSource tr = ui->transitions->itemData(i).value<OBSSource>();
-		if (!tr)
-			continue;
-
-		const char *trName = obs_source_get_name(tr);
-		if (strcmp(trName, name) == 0)
-			return tr;
-	}
+	if (trsNameUuid.find(name) == trsNameUuid.end())
+		return nullptr;
 
-	return nullptr;
+	return transitions[trsNameUuid[name]];
```

Comment:
```suggestion

    auto foundTransitionName = trsNameUuid.find(name);
    
    if (foundTransitionName == trsNameUuid.end()) {
        return nullptr;
    }

    try {
    	return transitions.at(foundTransitionName->second);
    } catch (const &std::out_of_range) {
        return nullptr;
    }
```

The subscript operator will implicitly _insert_ an element if no element with the specified key is found. The `at()` is non-mutating but will throw if given an invalid key.

Also the name is already reachable by the iterator returned by `find`, so there's no need to iterate over the map again via the subscript operator to yield it.

Reply:
Mutators removed